elasticsearch/sgconfig/config.yml

```diff
@@ -4,7 +4,7 @@ opendistro_security:
       xff:
         enabled: true
         remoteIpHeader: 'x-forwarded-for'
-        trustedProxies: '.*'
+        trustedProxies: '0\:0\:0\:0\:0\:0\:0\:1|127\.0\.0\.1'
         internalProxies: '.*'
```

Comment:
Remind me what 'internalProxies' means?  Does this need to be modified too?

Reply:
```
opendistro_security:
  dynamic:
    http:
      xff:
        enabled: true
        remoteIpHeader: 'x-forwarded-for'
        trustedProxies: '<regex-pattern>'
        internalProxies: '<regex-pattern>'
```
The remote address (InetAddress) from proxy request is matched against `internalProxies`.
`final String originalRemoteAddr = ((InetSocketAddress)request.getRemoteAddress()).getAddress().getHostAddress();`

If match fails, OPENDISTRO_SECURITY_XFF_DONE is false.
If `internalProxies` match succeeds, then the values in `X-Forwarded-For` header are matched against `trustedProxies`.
The matching is done in reverse order. Header values that match are put in `X-Forwarded-By` header, and values that do not match remain in `X-Forwarded-For` header. The `OPENDISTRO_SECURITY_XFF_DONE` is set to true as long as there is at least one value. 

`HTTPProxyAuthenticator` only checks if `OPENDISTRO_SECURITY_XFF_DONE` is true.
So we can use `internalProxies` to enforce the check that acess to ES is frol localhost